src/single.ts

```diff
@@ -160,7 +179,23 @@ export class RegionRatelimit extends Ratelimit<RegionContext> {
         [windowDuration],
       )) as number;
 
-      const success = usedTokensAfterUpdate <= tokens;
+      let success = usedTokensAfterUpdate <= tokens;
+
+      let remaining = Math.max(0, tokens - usedTokensAfterUpdate)
+      let remainingRates = 0;
+
+      if (rates && rate) {
+        const usedRatesAfterUpdate = (await ctx.redis.eval(
+          rateScript,
+          [key + ":" + "custom_limit"],
+          [rate, windowDuration],
+        )) as number;
+
+        success = success && usedRatesAfterUpdate <= rates
+
+        remainingRates = Math.max(0, rate - usedRatesAfterUpdate)
```

Comment:
Shouldn't this `rate` be `rates` instead? Because, let's say I set rates to 500, made a request then my `usedRatesAfterUpdate` is 250 and to get the remaining I have subtract `rates` from `usedRatesAfterUpdate`.
Otherwise, remaining rate always will be zero.

To test it you can do this:

 ```ts
 const ratelimit = new Ratelimit({
    redis: Redis.fromEnv(),
    limiter: Ratelimit.fixedWindow(10, "30s", 500),
  });
  const { success, limit, remaining, reset, remainingRates } =
    await ratelimit.limit(ip ?? "anonymous011", { rate: 250 });
  console.log({ success, limit, remaining, reset, remainingRates });

```

Reply:
Yes I refactored that and about to push,  still working on it